@here/harp-mapview/lib/VisibleTileSet.ts

```diff
@@ -916,89 +913,138 @@ export class VisibleTileSet {
-                            }
-                        }
-                    }
-                });
-                incompleteTiles = nextLevelCandidates;
+                if (
+                    searchDirection === SearchDirection.BOTH ||
+                    searchDirection === SearchDirection.DOWN
+                ) {
+                    this.findDown(tileKeyCode, displayZoomLevel, renderedTiles, dataSource);
+                }
             }
         });
     }
 
+    private findDown(
+        tileKeyCode: number,
+        displayZoomLevel: number,
+        renderedTiles: Map<number, Tile>,
+        dataSource: DataSource
+    ) {
+        const { offset, mortonCode } = TileOffsetUtils.extractOffsetAndMortonKeyFromKey(
+            tileKeyCode
+        );
+        const tileKey = TileKey.fromMortonCode(mortonCode);
+
+        const tilingScheme = dataSource.getTilingScheme();
+        for (const childTileKey of tilingScheme.getSubTileKeys(tileKey)) {
+            const childTileCode = TileOffsetUtils.getKeyForTileKeyAndOffset(childTileKey, offset);
+            const childTile = this.m_dataSourceCache.get(
+                childTileKey.mortonCode(),
+                offset,
+                dataSource
+            );
+
+            const nextLevelDiff = Math.abs(childTileKey.level - displayZoomLevel);
+            if (childTile !== undefined && childTile.hasGeometry) {
+                // childTile has geometry, so can be reused as fallback
+                renderedTiles.set(childTileCode, childTile);
+                childTile.levelOffset = nextLevelDiff;
+                continue;
+            }
+
+            // Recurse down until the max distance is reached.
+            if (nextLevelDiff < this.options.quadTreeSearchDistanceDown) {
+                this.findDown(childTileCode, displayZoomLevel, renderedTiles, dataSource);
+            }
+        }
+    }
+
+    /**
+     * Returns true if a tile was found in the cache which is a parent
+     * @param tileKeyCode Morton code of the current tile that should be searched for.
+     * @param displayZoomLevel The current zoom level of tiles that are to be displayed.
+     * @param renderedTiles The list of tiles that are shown to the user.
+     * @param checkedTiles Used to map a given code to a boolean which tells us if an ancestor is
+     * displayed or not.
+     * @param dataSource The provider of tiles.
+     * @returns Whether a parent tile exists.
+     */
+    private findUp(
+        tileKeyCode: number,
+        displayZoomLevel: number,
+        renderedTiles: Map<number, Tile>,
+        checkedTiles: Map<number, boolean>,
+        dataSource: DataSource
+    ): boolean {
+        const parentCode = TileOffsetUtils.getParentKeyFromKey(tileKeyCode);
+        // Check if another sibling has already added the parent.
+        if (renderedTiles.get(parentCode) !== undefined) {
+            return true;
```

Comment:
This is what fixes the bug mentioned, all siblings return true, which then continues, so the children of the siblings aren't searched.